src/Position.cpp

```diff
@@ -905,7 +905,7 @@ Key Position::key_after(Move m) const {
 
 bool Position::is_draw() const {  //--didn't understand this _ply_ parameter; deleting it.
 
-  if (st->rule50 > 99 && (!checkers() || MoveList<LEGAL>(*this).size()))
+  if (st->rule50 > 99 && (!checkers() || MoveList<LEGAL>(*this).size()) || popcount(pos.pieces() == 2)
       return true;
```

Comment:
That should probably be `popcount(pos.pieces()) == 2`.

Reply:
actually popcount(pos.pieces()) == 2)

Reply:
Yep :)